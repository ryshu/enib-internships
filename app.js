"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const express_1 = __importDefault(require("express"));
const body_parser_1 = __importDefault(require("body-parser"));
const compression_1 = __importDefault(require("compression")); // compresses requests
const lusca_1 = __importDefault(require("lusca"));
const dotenv_1 = __importDefault(require("dotenv"));
const express_flash_1 = __importDefault(require("express-flash"));
const path_1 = __importDefault(require("path"));
const method_override_1 = __importDefault(require("method-override"));
// Load environment variables from .env file, where API keys and passwords are configured
dotenv_1.default.config({ path: '.env' });
// Loggers
const logger_1 = __importDefault(require("./utils/logger"));
const error_1 = require("./utils/error");
const http_status_codes_1 = __importDefault(require("http-status-codes"));
// router and database
const route_1 = __importDefault(require("./configs/setup/route"));
const cas_1 = __importDefault(require("./configs/setup/cas"));
require("./configs/instances/database"); // Only import to setup
require("./configs/setup/database"); // Only import to setup
const express_session_1 = __importDefault(require("express-session"));
const handle_1 = require("./auth/cas/handle");
// Create Express server
const app = express_1.default();
exports.session = express_session_1.default({
    secret: process.env.SESSION_SECRET,
    resave: false,
    saveUninitialized: true,
});
// Express configuration
app.set('port', process.env.INTERNSHIP_ENIB_API_PORT || 3000);
app.use(compression_1.default());
app.use(body_parser_1.default.json({ limit: '1mb' }));
app.use(body_parser_1.default.urlencoded({ extended: true, limit: '1mb' }));
app.use(express_flash_1.default());
app.use(lusca_1.default.xframe('SAMEORIGIN'));
app.use(lusca_1.default.xssProtection(true));
app.use(exports.session);
// CAS Setup
app.get('/logout', cas_1.default.logout); // Logout pass
app.use('/', cas_1.default.bounce, // CAS bounce to connect user if isn't
(req, _res, next) => {
    if (!req.session.info) {
        // If no session info, handle connection
        handle_1.handleConnection(req)
            .then(() => next())
            .catch((e) => next(e));
    }
    else {
        // else, pass
        next();
    }
}, express_1.default.static(path_1.default.join(__dirname, 'public'), { maxAge: 0 }));
app.use(route_1.default);
/**
 * Error handler setup
 */
app.use(method_override_1.default());
app.use((err, _req, res, _next) => {
    if (err instanceof error_1.APIError) {
        return res.status(err.status).json(err);
    }
    else {
        logger_1.default.error(err);
        return res.status(http_status_codes_1.default.INTERNAL_SERVER_ERROR).send(`Internal server error`);
    }
});
exports.default = app;
//# sourceMappingURL=app.js.map