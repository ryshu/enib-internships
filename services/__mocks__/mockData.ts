export function defaultBusiness() {
    const VALID_BUSINESS: IBusinessEntity = {
        name: 'test',
        country: 'France',
        city: 'Brest',
        postalCode: '29200',
        address: 'TEST',
    };
    return VALID_BUSINESS;
}

export function defaultFiles() {
    const VALID_FILE: IFileEntity = {
        name: 'fichier',
        size: 150,
        type: 'john.doe@enib.fr',
        path: 'S10',
    };
    return VALID_FILE;
}

export function defaultInternships() {
    const VALID_INTERNSHIP: IInternshipEntity = {
        subject: 'test',
        description: 'Stage',
        country: 'France',
        city: 'Brest',
        postalCode: '29280',
        address: 'TEST',
        isLanguageCourse: true,
        isValidated: false,
    };
    return VALID_INTERNSHIP;
}

export function defaultMentoringPropositions() {
    const VALID_MENTORING_PROPOSITION: IMentoringPropositionEntity = {
        comment: 'test',
    };
    return VALID_MENTORING_PROPOSITION;
}

export function defaultStudents() {
    const VALID_STUDENT: IStudentEntity = {
        firstName: 'John',
        lastName: 'Doe',
        email: 'john.doe@enib.fr',
        semester: 'S10',
    };
    return VALID_STUDENT;
}

export function defaultInternshipTypes() {
    const VALID_INTERNSHIP_TYPES: IInternshipTypeEntity = {
        label: 'test',
    };
    return VALID_INTERNSHIP_TYPES;
}
