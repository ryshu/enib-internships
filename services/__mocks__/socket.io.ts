import SubSocket from './subsocket';

export class SocketServer {
    public of(_nsp: string | RegExp): any {
        return new SubSocket();
    }

    public close(fn?: () => void): void {
        fn();
    }

    public on(_event: 'connection', _listener: (socket: any) => void): any {
        return;
    }

    public emit(_event: string, ..._args: any[]): any {
        return jest.fn();
    }

    public to() {
        return this;
    }
}

export default function() {
    return new SocketServer();
}
