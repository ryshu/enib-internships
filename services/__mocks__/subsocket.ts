export default class SubSocket {
    public emit ( ..._args: any[]): void {
        return;
    }
}
