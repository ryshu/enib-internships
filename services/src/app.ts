import express, { Request, Response, NextFunction } from 'express';
import bodyParser from 'body-parser';
import cors from 'cors';
import compression from 'compression'; // compresses requests
import lusca from 'lusca';
import dotenv from 'dotenv';
import flash from 'express-flash';
import path from 'path';
import methodOverride from 'method-override';

// Load environment variables from .env file, where API keys and passwords are configured
dotenv.config({ path: '.env' });

// Loggers
import logger from './utils/logger';
import { APIError } from './utils/error';
import httpStatus from 'http-status-codes';

// router and database
import router from './configs/setup/route';
import './configs/instances/database'; // Only import to setup
import './configs/setup/database'; // Only import to setup

// Create Express server
const app = express();

// Express configuration
app.set('port', process.env.INTERNSHIP_ENIB_API_PORT || 3000);
app.use(compression());
app.use(bodyParser.json({ limit: '1mb' }));
app.use(bodyParser.urlencoded({ extended: true, limit: '1mb' }));
app.use(flash());
app.use(lusca.xframe('SAMEORIGIN'));
app.use(lusca.xssProtection(true));

app.use(express.static(path.join(__dirname, 'public'), { maxAge: 31557600000 }));

const corsOptions = {
    origin: 'http://localhost:4200',
    optionsSuccessStatus: 200,
};
app.use(cors(corsOptions));

app.use(router);

/**
 * Error handler setup
 */
app.use(methodOverride());
app.use((err: APIError | Error, _req: Request, res: Response, _next: NextFunction) => {
    if (err instanceof APIError) {
        return res.status(err.status).json(err);
    } else {
        logger.error(err);
        return res.status(httpStatus.INTERNAL_SERVER_ERROR).send(`Internal server error`);
    }
});

export default app;
