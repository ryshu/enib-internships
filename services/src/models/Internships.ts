import * as Sequelize from 'sequelize';

import database from '../configs/instances/database';

import Businesses from './Businesses';
import InternshipTypes from './InternshipTypes';

class Internships extends Sequelize.Model implements IInternshipEntity {
    public static associations: {
        category: Sequelize.Association<Internships, InternshipTypes>;
        business: Sequelize.Association<Internships, Businesses>;
    };

    public id!: number; // Note that the `null assertion` `!` is required in strict mode.

    // Data
    public subject: string;
    public description: string;

    // Localisation
    public country: string;
    public city: string;
    public postalCode: string;
    public address: string;
    public additional?: string;

    // State
    public isInternshipAbroad: boolean;
    public isValidated: boolean;

    // timestamps!
    public readonly createdAt!: Date;
    public readonly updatedAt!: Date;

    public getBusiness: Sequelize.BelongsToGetAssociationMixin<Businesses>;
    public setBusiness: Sequelize.BelongsToSetAssociationMixin<Businesses, Businesses['id']>;
    public createBusiness: Sequelize.BelongsToCreateAssociationMixin<IBusinessEntity>;

    public getCategory: Sequelize.BelongsToGetAssociationMixin<InternshipTypes>;
    public setCategory: Sequelize.BelongsToSetAssociationMixin<
        InternshipTypes,
        InternshipTypes['id']
    >;
    public createCategory: Sequelize.BelongsToCreateAssociationMixin<IInternshipTypeEntity>;

    public readonly business?: Businesses | Businesses['id'];
    public readonly category?: InternshipTypes | InternshipTypes['id'];
}

Internships.init(
    {
        id: {
            type: Sequelize.DataTypes.INTEGER.UNSIGNED,
            autoIncrement: true,
            primaryKey: true,
        },

        // Data
        subject: {
            type: new Sequelize.DataTypes.STRING(256),
            allowNull: false,
        },
        description: {
            type: new Sequelize.DataTypes.TEXT(),
            allowNull: true,
        },

        // Localisation
        country: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
        city: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
        postalCode: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
        address: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
        additional: {
            type: new Sequelize.DataTypes.STRING(),
            allowNull: true,
        },

        // State
        isInternshipAbroad: {
            type: Sequelize.DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
        },
        isValidated: {
            type: Sequelize.DataTypes.BOOLEAN,
            allowNull: false,
            defaultValue: false,
        },
    },
    {
        tableName: 'internships',
        sequelize: database,
        defaultScope: {
            attributes: { exclude: ['businessId', 'categoryId'] },
        },
    },
);

export default Internships;
