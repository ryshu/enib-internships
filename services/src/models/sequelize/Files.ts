import * as Sequelize from 'sequelize';
import Internships from './Internships';
import database from '../../configs/instances/database';

import { IInternshipEntity } from '../../declarations';

class Files extends Sequelize.Model {
    public static associations: {
        internship: Sequelize.Association<Files, Internships>;
    };
    public id!: number; // Note that the `null assertion` `!` is required in strict mode.

    public name: string;
    public type: string;
    public path: string;

    // timestamps!
    public readonly createdAt!: Date;
    public readonly updatedAt!: Date;

    // Internship
    public getInternship: Sequelize.BelongsToGetAssociationMixin<Internships>;
    public setInternship: Sequelize.BelongsToSetAssociationMixin<Internships, Internships['id']>;
    public createInternship: Sequelize.BelongsToCreateAssociationMixin<IInternshipEntity>;

    public readonly internship?: Internships;
}

Files.init(
    {
        id: {
            type: Sequelize.DataTypes.INTEGER.UNSIGNED,
            autoIncrement: true,
            primaryKey: true,
        },
        name: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
        type: {
            type: new Sequelize.DataTypes.STRING(20),
            allowNull: false,
        },
        path: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
    },
    {
        tableName: 'files',
        sequelize: database,
        timestamps: true,
        paranoid: true,
        defaultScope: {
            attributes: { exclude: ['internshipId'] },
        },
    },
);

export default Files;
