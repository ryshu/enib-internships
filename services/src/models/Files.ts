import * as Sequelize from 'sequelize';
import Internships from './Internships';
import database from '../configs/instances/database';

class Files extends Sequelize.Model implements IFileEntity {
    public static associations: {
        internship: Sequelize.Association<Files, Internships>;
    };
    public id!: number; // Note that the `null assertion` `!` is required in strict mode.

    public name: string;
    public size: number;
    public type: string;
    public path: string;

    // timestamps!
    public readonly createdAt!: Date;
    public readonly updatedAt!: Date;

    // Internship
    public getInternship: Sequelize.BelongsToGetAssociationMixin<Internships>;
    public setInternship: Sequelize.BelongsToSetAssociationMixin<Internships, Internships['id']>;
    public createInternship: Sequelize.BelongsToCreateAssociationMixin<IInternshipEntity>;

    public readonly internship?: Internships[];
}

Files.init(
    {
        id: {
            type: Sequelize.DataTypes.INTEGER.UNSIGNED,
            autoIncrement: true,
            primaryKey: true,
        },
        name: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
        size: {
            type: Sequelize.DataTypes.INTEGER.UNSIGNED,
            allowNull: false,
        },
        type: {
            type: new Sequelize.DataTypes.STRING(20),
            allowNull: false,
        },
        path: {
            type: new Sequelize.DataTypes.STRING(128),
            allowNull: false,
        },
    },
    {
        tableName: 'files',
        sequelize: database,
        defaultScope: {
            attributes: { exclude: ['internshipId'] },
        },
    },
);

export default Files;
