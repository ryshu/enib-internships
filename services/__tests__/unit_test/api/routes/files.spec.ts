import request from 'supertest';

// Include app for super agent, not server
import app from '../../../../src/app';

// Include database connection
import dbSetup from '../../../../src/configs/setup/database';

// Import model for pre-operation before asserting API methods
import Files from '../../../../src/models/Files';
import Internships from '../../../../src/models/Internships';

import { defaultFiles, defaultInternships } from '../../../../__mocks__/mockData';

beforeAll((done) => {
    dbSetup.then(() => done()).catch((e) => done(e));
});

describe('GET /files', () => {
    beforeEach(async () => {
        // Remove all
        await Files.destroy({ where: {} });
    });

    it('NoFile_204', async () => {
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files`,
        );
        expect(RESPONSE.status).toBe(204);
    });

    it('BadRequest_400', async () => {
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files?limit=Nan`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('Files_200', async () => {
        const VALID_FILE = defaultFiles();

        await Files.create(VALID_FILE);
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files`,
        );
        expect(RESPONSE.status).toBe(200);
        expect(Array.isArray(RESPONSE.body.data)).toBeTruthy();
        expect(RESPONSE.body).toMatchSnapshot({
            data: [
                {
                    createdAt: expect.any(String),
                    updatedAt: expect.any(String),
                },
            ],
        });
    });
});

describe('POST /files', () => {
    it('NoBody_400', async () => {
        const RESPONSE = await request(app).post(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('ValidFiles_200', async () => {
        const VALID_FILE = defaultFiles();

        const RESPONSE = await request(app)
            .post(`/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files`)
            .send(VALID_FILE);
        expect(RESPONSE.status).toBe(200);
        expect(RESPONSE.body).toMatchSnapshot({
            createdAt: expect.any(String),
            updatedAt: expect.any(String),
            id: expect.any(Number),
        });
    });
});

describe('GET /files/:id', () => {
    beforeEach(async () => {
        // Remove all
        await Files.destroy({ where: {} });
    });

    it('NoFile_204', async () => {
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/10`,
        );
        expect(RESPONSE.status).toBe(204);
    });

    it('BadRequest_400', async () => {
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/{falseEncoding}`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('Files_200', async () => {
        const VALID_FILE = defaultFiles();

        const CREATED = await Files.create(VALID_FILE);
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED.id}`,
        );
        expect(RESPONSE.status).toBe(200);
        expect(RESPONSE.body).toMatchSnapshot({
            createdAt: expect.any(String),
            updatedAt: expect.any(String),
            id: expect.any(Number),
        });
    });
});

describe('PUT /files/:id', () => {
    beforeEach(async () => {
        // Remove all
        await Files.destroy({ where: {} });
    });

    it('NoFile_204', async () => {
        const RESPONSE = await request(app).put(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/10`,
        );
        expect(RESPONSE.status).toBe(204);
    });

    it('BadRequest_400', async () => {
        const RESPONSE = await request(app).put(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/{falseEncoding}`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('Files_200_UpdateAllData', async () => {
        const VALID_FILE = defaultFiles();

        const CREATED = await Files.create(VALID_FILE);

        // Change some data on VALID_FILE
        VALID_FILE.size = 250;

        const RESPONSE = await request(app)
            .put(`/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED.id}`)
            .send(VALID_FILE);
        expect(RESPONSE.status).toBe(200);
        expect(RESPONSE.body).toMatchSnapshot({
            createdAt: expect.any(String),
            updatedAt: expect.any(String),
            id: expect.any(Number),
        });
    });

    it('Files_200_NotAnyDataUpdated', async () => {
        const VALID_FILE = defaultFiles();

        const CREATED = await Files.create(VALID_FILE);

        const RESPONSE = await request(app)
            .put(`/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED.id}`)
            .send({});
        expect(RESPONSE.status).toBe(200);
        expect(RESPONSE.body).toMatchSnapshot({
            createdAt: expect.any(String),
            updatedAt: expect.any(String),
            id: expect.any(Number),
        });
    });
});

describe('DELETE /files/:id', () => {
    beforeEach(async () => {
        // Remove all
        await Files.destroy({ where: {} });
    });

    it('NoFile_200', async () => {
        const RESPONSE = await request(app).delete(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/10`,
        );
        expect(RESPONSE.status).toBe(200);
    });

    it('BadRequest_400', async () => {
        const RESPONSE = await request(app).delete(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/{falseEncoding}`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('Files_200', async () => {
        const VALID_FILE = defaultFiles();

        const CREATED = await Files.create(VALID_FILE);

        const RESPONSE = await request(app).delete(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED.id}`,
        );
        expect(RESPONSE.status).toBe(200);
    });
});

describe('GET /files/:id/internships', () => {
    beforeEach(async () => {
        // Remove all
        await Files.destroy({ where: {} });
    });

    it('NoInternship_204', async () => {
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/10/internships`,
        );
        expect(RESPONSE.status).toBe(204);
    });

    it('BadRequest_400', async () => {
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/{falseEncoding}/internships`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('Files_200_NoLinkedData', async () => {
        const VALID_FILE = defaultFiles();

        const CREATED = await Files.create(VALID_FILE);
        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED.id}/internships`,
        );
        expect(RESPONSE.status).toBe(200);
        expect(RESPONSE.body).toEqual({});
    });

    it('Files_200_WithLinkedData', async () => {
        const VALID_FILE = defaultFiles();
        const VALID_INTERNSHIP = defaultInternships();

        const CREATED_INTERNSHIP = await Internships.create(VALID_INTERNSHIP);
        let CREATED_FILE = await Files.create(VALID_FILE);

        await CREATED_FILE.setInternship(CREATED_INTERNSHIP.id);
        CREATED_FILE = await Files.findByPk(CREATED_FILE.id);

        const RESPONSE = await request(app).get(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED_FILE.id}/internships`,
        );
        expect(RESPONSE.status).toBe(200);
        expect(RESPONSE.body).toMatchSnapshot({
            createdAt: expect.any(String),
            updatedAt: expect.any(String),
        });
    });
});

describe('POST /files/:id/internships/:internship_id/link', () => {
    beforeEach(async () => {
        // Remove all
        await Files.destroy({ where: {} });
    });

    it('NoInternship_204', async () => {
        const RESPONSE = await request(app).post(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/10/internships/20/link`,
        );
        expect(RESPONSE.status).toBe(204);
    });

    it('BadRequest_400_WrongID', async () => {
        const RESPONSE = await request(app).post(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/{falseEncoding}/internships/10/link`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('BadRequest_400_WrongBusinessID', async () => {
        const RESPONSE = await request(app).post(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/10/internships/{falseEncoding}/link`,
        );
        expect(RESPONSE.status).toBe(400);
    });

    it('Files_204_NoBusiness', async () => {
        // In this case, we check if link a existing files and an unexisting internships work
        const VALID_FILE = defaultFiles();

        const CREATED = await Files.create(VALID_FILE);
        const RESPONSE = await request(app).post(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED.id}/internships/20/link`,
        );
        expect(RESPONSE.status).toBe(204);
    });

    it('Files_200_WithBusiness', async () => {
        const VALID_INTERNSHIP = defaultInternships();
        const VALID_FILE = defaultFiles();

        const CREATED_INTERNSHIP = await Internships.create(VALID_INTERNSHIP);
        let CREATED_FILE = await Files.create(VALID_FILE);

        const RESPONSE = await request(app).post(
            `/api/${process.env.INTERNSHIP_ENIB_API_VERSION}/files/${CREATED_FILE.id}/internships/${CREATED_INTERNSHIP.id}/link`,
        );

        // Should answer 200
        expect(RESPONSE.status).toBe(200);

        // check if category and internship are linked
        CREATED_FILE = await Files.findByPk(CREATED_FILE.id, {
            include: [{ model: Internships, as: 'internship' }],
        });

        const data = JSON.parse(JSON.stringify(CREATED_FILE.internship)) as any;

        expect(CREATED_FILE.internship).toBeTruthy();
        expect(data).toMatchSnapshot({
            createdAt: expect.any(String),
            updatedAt: expect.any(String),
        });
    });
});
