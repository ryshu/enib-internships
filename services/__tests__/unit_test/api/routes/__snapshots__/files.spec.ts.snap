// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /files Files_200 1`] = `
Object {
  "data": Array [
    Object {
      "createdAt": Any<String>,
      "id": 1,
      "name": "fichier",
      "path": "S10",
      "size": 150,
      "type": "john.doe@enib.fr",
      "updatedAt": Any<String>,
    },
  ],
  "length": 1,
  "max": 1,
  "page": 1,
}
`;

exports[`GET /files/:id Files_200 1`] = `
Object {
  "createdAt": Any<String>,
  "id": Any<Number>,
  "internship": null,
  "name": "fichier",
  "path": "S10",
  "size": 150,
  "type": "john.doe@enib.fr",
  "updatedAt": Any<String>,
}
`;

exports[`GET /files/:id/internships Files_200_WithLinkedData 1`] = `
Object {
  "additional": null,
  "address": "TEST",
  "city": "Brest",
  "country": "France",
  "createdAt": Any<String>,
  "description": "Stage",
  "endAt": 20191012,
  "id": 1,
  "isInternshipAbroad": true,
  "isValidated": false,
  "postalCode": "29280",
  "startAt": 20191012,
  "subject": "test",
  "updatedAt": Any<String>,
}
`;

exports[`POST /files ValidFiles_200 1`] = `
Object {
  "createdAt": Any<String>,
  "id": Any<Number>,
  "name": "fichier",
  "path": "S10",
  "size": "150",
  "type": "john.doe@enib.fr",
  "updatedAt": Any<String>,
}
`;

exports[`POST /files/:id/internships/:internship_id/link Files_200_WithBusiness 1`] = `
Object {
  "additional": null,
  "address": "TEST",
  "city": "Brest",
  "country": "France",
  "createdAt": Any<String>,
  "description": "Stage",
  "endAt": 20191012,
  "id": 2,
  "isInternshipAbroad": true,
  "isValidated": false,
  "postalCode": "29280",
  "startAt": 20191012,
  "subject": "test",
  "updatedAt": Any<String>,
}
`;

exports[`PUT /files/:id Files_200_NotAnyDataUpdated 1`] = `
Object {
  "createdAt": Any<String>,
  "id": Any<Number>,
  "name": "fichier",
  "path": "S10",
  "size": 150,
  "type": "john.doe@enib.fr",
  "updatedAt": Any<String>,
}
`;

exports[`PUT /files/:id Files_200_UpdateAllData 1`] = `
Object {
  "createdAt": Any<String>,
  "id": Any<Number>,
  "name": "fichier",
  "path": "S10",
  "size": "250",
  "type": "john.doe@enib.fr",
  "updatedAt": Any<String>,
}
`;
